miden-lib/asm/miden/kernels/tx/note.masm

```diff
@@ -87,6 +87,50 @@ export.get_inputs_info
     # => [NOTE_INPUTS_HASH, num_inputs]
 end
 
+#! Returns the note script hash of the note currently being processed. Panics if a note is not
+#! being processed.
+#!
+#! Inputs: []
+#! Outputs: [NOTE_SCRIPT_HASH]
+#!
+#! - NOTE_INPUTS_HASH is the note inputs hash of the note currently being processed.
+export.get_script_root
```

Comment:
Why do we need this and `get_note_serial_number` procedures? It might be good to have the in general, but we should do this in a different PR.

Reply:
Because I didn't want to store the note_args in memory but retrieve them from the advice provider if necessary. That means a user could get the note args like 

```
exec.note::get_note_id
#=> [NOTE_ID]

exec.note::get_note_args
#=> [NOTE_ARGS]
```

If we put the note args directly on the stack when processing a note, we don't need those new procedures.